dashboard/src/components/dashboard/AccountMenu.tsx

```diff
@@ -23,21 +21,19 @@ function AccountMenuContent({
   const user = useUserData();
   const router = useRouter();
   const [clicked, setClicked] = useState(false);
-  const { setWorkspaceContext } = useWorkspaceContext();
-  const { setUserContext } = useUserDataContext();
   const { handleClose } = useDropdown();
+  const client = useApolloClient();
 
   return (
-    <div className="relative grid w-account grid-flow-row gap-5 p-6">
+    <div className="relative grid grid-flow-row gap-5 p-6 w-account">
       <Button
         variant="borderless"
         color="secondary"
-        className="absolute top-6 right-4 grid grid-flow-col items-center gap-1 self-start font-medium"
+        className="absolute grid items-center self-start grid-flow-col gap-1 font-medium top-6 right-4"
         onClick={async () => {
-          setWorkspaceContext(emptyWorkspace());
-          setUserContext({ workspaces: [] });
+          await router.push('/signin');
```

Comment:
I think we shouldn't change the order of the sign-out and the router.push, because the sign-in page would try to redirect us to the dashboard's home page when it sees that we are signed in.

Reply:
OK after testing the re-order:  
  
This is what works the best and doesn't generate a flicker when signing out nor the cache errors:  
  
\`\`\`ts  
router.push('/signin');

**await** nhost.auth.signOut();

**await** client.resetStore();  
\`\`\`

This one generates the `workspace_member` field not found:

\`\`\`ts  
router.push('/signin');

**await** nhost.auth.signOut();

**await** client.resetStore();  
\`\`\`

This one `expecting a value for non-nullable variable: appId`:

```typescript
await client.resetStore();
nhost.auth.signOut();
await router.push('/signin');
```

Reply:
The first and second are looking identical to me, but thanks for the investigation. I see what you mean. Go with the one with the best result!